From 1221a60368431d2fe79c0a0b65199bef590d145c Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 19 Dec 2024 10:05:32 +0000
Subject: [PATCH] dts: bcm2712-ds: Restore the VGIC interrupt

The upstream bcm2712.dtsi now includes the GIC, but it is missing the
interrupt declaration - the VGIC interrupt which is required by KVM.
Add it back in bcm2712-dt.dtsi.

Fixes: https://github.com/raspberrypi/linux/issues/6550

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

--- a/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-ds.dtsi
@@ -613,6 +613,11 @@
 	};
 };
 
+&gicv2 {
+	interrupts = <GIC_PPI 9 (GIC_CPU_MASK_SIMPLE(4) |
+				 IRQ_TYPE_LEVEL_HIGH)>;
+};
+
 &aon_intr {
 	status = "disabled";
 };
